### ISSUE TITLE
possible regression in JXL save from ICC profile?

### ISSUE THREAD

**Issue by richardmonette** [libvips/libvips#2872]
@jcupitt @f1ac is it possible https://github.com/libvips/libvips/pull/2815 caused a regression in terms of the compressed filesize?

```
> vips jxlsave photo-1595841696677-6489ff3f8cd1.jpeg photo-1595841696677-6489ff3f8cd1_direct.jxl
> vips -v
vips-8.12.2-Tue Jan 25 09:34:32 UTC 2022
> ll photo-1595841696677-6489ff3f8cd1_direct.jxl
-rw-r--r--  1 richardmonette  staff  337549 17 Jun 12:49 photo-1595841696677-6489ff3f8cd1_direct.jxl
```

when I run against 8.12.2 it appears saving a JXL is about 338kb whereas when I build and run against https://github.com/libvips/libvips/commit/6260a37136973f3d231d0e2c28af6190126c664e (from https://github.com/libvips/libvips/archive/6260a37136973f3d231d0e2c28af6190126c664e.tar.gz) the filesize jumps to about 931kb.

**Comment by richardmonette** [comment]
Here is an example with the original jpeg, and then the 😄 JXL from 8.12 and the 😢 JXL from the SHA above: [jxl_icc_regression_example.zip](https://github.com/libvips/libvips/files/8929412/jxl_icc_regression_example.zip)

**Comment by f1ac** [comment]
Sounds like uses_original_profile = true causes this.  This setting forces libjxl to not use XYB encoding, which is necessary for lossless encoding, but is less effective. IMHO this should be set conditionally, out->uses_original_profile = in->lossless.

**Comment by f1ac** [comment]
BTW uses_original_profile does not seem to have anything to do with ICC profiles, just a confusing name for lossy XYB colourspace toggle.

**Comment by jcupitt** [comment]
.... I added some tests too. It checks lossy and lossless mode, with and without an ICC profile, plus 8 bit, 16 bit, and float. It also checks lossy is much smaller than lossless.

Any more testing very welcome!

**Comment by jcupitt** [comment]
@richardmonette I tried your test image after these changes:

```
$ vips copy photo-1595841696677-6489ff3f8cd1.jpeg x.jxl
$ ls -l photo-1595841696677-6489ff3f8cd1.jpeg x.jxl
-rw-r--r-- 1 john john 1765625 Jun 16 14:46 photo-1595841696677-6489ff3f8cd1.jpeg
-rw-rw-r-- 1 john john  337774 Jun 18 16:42 x.jxl
```

Thanks for taking the time to try this and make the report.

**Comment by jcupitt** [comment]
... this fix is in 8.13-rc1, please open a new issue if it's still broken.

https://github.com/libvips/libvips/releases/tag/v8.13.0-rc1
